lib/packwerk/application_validator.rb

```diff
@@ -176,7 +176,7 @@ def check_inflection_file
       Packwerk::Inflections::Custom.new(inflections_file).apply_to(test_inflections)
 
       results = %i(plurals singulars uncountables humans acronyms).map do |type|
-        expected = ActiveSupport::Inflector.inflections.public_send(type).to_set
+        expected = Packwerk::Inflector.from_file(inflections_file).inflections.public_send(type).to_set
```

Comment:
What is `Packwerk::ApplicationValidator#check_inflection_file` supposed to be checking? I don’t understand under what circumstances this check is intended to fail, which means I don’t know enough to say whether this change is right.

Reply:
Perhaps @exterm can give a better response since you wrote it? I am not quite sure either. From what I understand, we've previously used `ActiveSupport::Inflector` as the inflector, and now just switching over to the Packwerk's version.

Reply:
Ah, this change is wrong - sorry for not writing the code in a more obvious way!

The purpose of this check is to compare the inflections that are configured for packwerk with the actual inflections present in the application. So this line really should not be changed.

Reply:
Wait no I am confused

Reply:
Yeah, this whole check is super confusing.

But suffice it to say, my explanation is correct, the previous version was correct, this change breaks it.

Reply:
Further up we have 

```
      test_inflections = ActiveSupport::Inflector::Inflections.new
      Packwerk::Inflections::Default.apply_to(test_inflections)
      Packwerk::Inflections::Custom.new(inflections_file).apply_to(test_inflections)
```

and then down here

```
expected = ActiveSupport::Inflector.inflections.public_send(type).to_set
actual = test_inflections.public_send(type).to_set
```

So `expected` is the application's inflections and `actual` is packwerk's inflections.

Reply:
It can be simplified to (pseudocode)

```
expected = application_inflections = ActiveSupport::Inflector.inflections.public_send(type).to_set
actual = packwerk_inflections = Packwerk::Inflector.from_file(configuration.inflections_file).inflections.public_send(type).to_set
```